src/gretel_trainer/relational/backup.py

```diff
@@ -10,6 +10,7 @@
 @dataclass
 class BackupRelationalDataTable:
     primary_key: List[str]
+    invented_table_metadata: Optional[dict[str, str]] = None
```

Comment:
Is the word "invented" used here to potentially capture wider set of _inventions_ that lead to creating extra tables? For now can this be thought of as _nested_ tables?

Reply:
After more reading through this, it looks like a table is _invented_ if:
- it was flattened, or
- it was created from nested data

Reply:
Ironically, I think your two bullet points could actually be two ways of describing a single situation—lifting the key-value pairs of a JSON dictionary to standalone columns sounds like "flattening nested data" to me 😄 

I feel like the word "invented" makes it clear that this is not a table that a user provided but one that was created internally; meanwhile, if we used "nested" I almost anticipate wondering "wait, is this the source table that does have nested data, or is it the table we created that has **de-nested** data?". Invented also feels a little future-proof in case we need to create more non-user-source tables for some reason.

Reply:
Oh, I see, yeah, this makes sense, thanks for the explanation!